### PR TITLE
Make plugin task first-class citizen

### PR DESCRIPTION
**Pull request by honnix**
This is the same as #268, which was not ready to get merged and got reverted later.

# TL;DR
Create a dedicated Plugin Task instead of abusing Runnable Task.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
* ~~Manually pick proto change from unmerged PR https://github.com/flyteorg/flyte/pull/4107~~ We can go back to this later, for now I have removed this part to make this PR shippable
* Now we have a proper Plugin Task so users will no longer to have implement Runnable Task with an a `run` returning `null`

Note that it is still supported to use Runnable Task to implement tasks that are handled by backend plugin, for backward compatibility.

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by honnix** [comment]
I dropped the `isSyncPlugin` field from this PR because the upstream might take some time.

**Review by andresgomezfrr — APPROVED** [review]
